project_ideas/app/views.py

```diff
@@ -0,0 +1,195 @@
+            authenticated_user = backend.do_auth(access_token, user=user)
+        except Exception as error:
+            return Response({
+                "error": {
+                    "access_token":"Invalid token",
+                    "details":str(error)
+                }
+            }, status=status.HTTP_400_BAD_REQUEST)
+
+
+        if authenticated_user and authenticated_user.is_active:     
+            #generate Token for authtication
+            token, _ = Token.objects.get_or_create(user=user)
+            response = {
+                "id": user.id,
+                "email": authenticated_user.email,
+                "username": authenticated_user.username,                
+                "token":token.key
+            }
+            return Response(status=status.HTTP_200_OK, data=response)
+
+
+class SocialLogoutView(APIView):
+
+    def get(self, request, format=None):
+        user = request.user
+        response = {
+            "message":"User logged out", 
+            "Details":{
+                "id": user.id,
+                "username": user.username,
+                "email": user.email
+            }}
+        request.user.auth_token.delete()
+        return Response(response, status=status.HTTP_200_OK)
+            
+
+class PostIdeaView(APIView):
+    permission_classes = (AllowAny,)
+
+    def post(self, request):
+        serializer = IdeaSerializer(data = request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response({"message":serializer.data}, status=status.HTTP_200_OK)
+        else:
+            return Response({"message": "Invalid Idea Posted"}, status=status.HTTP_400_BAD_REQUEST)
+
+
+class PublishedIdeasView(APIView):
+    permission_classes = (AllowAny,)
+
+    def get(self, request):
+        ideas = list(Idea.objects.filter(is_reviewed=1))
+        serializers = (IdeaSerializer(ideas, many=True))
+        if len(ideas) == 0:
+            return Response({"message":"No Published Ideas found"}, status=status.HTTP_204_NO_CONTENT)
+        return Response({"message":serializers.data}, status=status.HTTP_200_OK)
+
+
+class ViewIdea(APIView):
+    permission_classes = (AllowAny,)
+
+    def get(self, request, pk):
+        try:
+            idea = Idea.objects.get(id=pk)
+            serializer = IdeaSerializer(idea)
+            return Response({"message":serializer.data}, status=status.HTTP_200_OK)
+        except Idea.DoesNotExist:
+            return Response({"message":"Idea not found or Invalid id number"}, status=status.HTTP_204_NO_CONTENT)
+
+
+class VoteView(APIView):
+
+    def post(self, request):
+        req_data = request.data
+        user = request.user
+        try:
+            idea = Idea.objects.get(id=req_data["idea_id"])
+        except Idea.DoesNotExist:
+            return Response({"message":"Invalid Idea Id"}, status=status.HTTP_400_BAD_REQUEST)
+        if idea.is_reviewed==0 or idea.is_reviewed==2:
```

Comment:
Add comments here as no one knows what is 1 and 2

Reply:
Added comments